Signals/DataAccess/Repositories/SignalsDataRepository.cs

```diff
@@ -49,12 +49,23 @@ public IEnumerable<Datum<T>> GetData<T>(Signal signal, DateTime fromIncludedUtc,
         public void SetData<T>(IEnumerable<Datum<T>> data)
         {
             var concreteDatumType = GetConcreteDatumType<T>();
+            var oldData = data.Count() > 0
+                       ? GetData<T>(data.First().Signal, data.First().Timestamp, data.Last().Timestamp.AddSeconds(1)).ToDictionary(d => d.Timestamp)
```

Comment:
I'd have to create that enumerator somehow. True, that `AddSeconds` is a little bit ugly, but we won't go below seconds, and even if - method will still work, it will just load too many samples